API/common/utils.py

```diff
@@ -518,3 +518,14 @@ def read_objects_from_libs(libpath, liblist):
         objlist.extend(output.splitlines())
 
     return objlist
+
+
+def read_port_from_url(url):
+    '''
+    Parse URL and return with the port number
+    '''
+    p = '(?:http.*://)?(?P<host>[^:/ ]+).?(?P<port>[0-9]*).*'
+
+    m = re.search(p, url)
+
+    return m.group('port')
```

Comment:
If `port` is not defined with the host, there won't be match. An error message or a default port can be useful.